File: r/src/R/ai/h2o/sparkling/H2OConf.R
if (!exists("SharedBackendConf")) source(file.path("R", "SharedBackendConf.R"))
if (!exists("ExternalBackendConf")) source(file.path("R", "ExternalBackendConf.R"))
if (!exists("InternalBackendConf")) source(file.path("R", "InternalBackendConf.R"))

#' @export H2OConf
H2OConf <- setRefClass("H2OConf", fields = list(jconf = "ANY"),
                       contains = c("SharedBackendConf", "ExternalBackendConf", "InternalBackendConf"), methods = list(
    initialize = function() {
        sc <- spark_connection_find()[[1]]
        .self$jconf <- invoke_new(sc, "ai.h2o.sparkling.H2OConf")
    },

    set = function(option, value) { invoke(jconf, "set", option, value); .self },

    get = function(option) { invoke(jconf, "get", option) }
))

File: py/src/ai/h2o/sparkling/H2OConf.py
from ai.h2o.sparkling.ExternalBackendConf import ExternalBackendConf
from ai.h2o.sparkling.Initializer import Initializer
from ai.h2o.sparkling.InternalBackendConf import InternalBackendConf
from ai.h2o.sparkling.SharedBackendConf import SharedBackendConf
from pyspark.sql import SparkSession


class H2OConf(SharedBackendConf, InternalBackendConf, ExternalBackendConf):
    def __init__(self):
        try:
            Initializer.load_sparkling_jar()
            _jvm = SparkSession._instantiatedSession.sparkContext._jvm
            self._jconf = _jvm.ai.h2o.sparkling.H2OConf()
        except:
            raise
